src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapFandom.java

```diff
@@ -0,0 +1,190 @@
+
+        videoFandomPage().verifyFandomPageOpend(videoFanTakeover);
+    }
+
+    @InBrowser(
+            browser = Browser.CHROME,
+            emulator = Emulator.GOOGLE_NEXUS_5
+    )
+    @Test(
+            dataProviderClass = FandomAdsDataProvider.class,
+            dataProvider = "fandomVuapPageMobile",
+            groups = {"AdsVuapFandomMobile", "AdsVuapVideoClosesWhenTapCloseButtonFandomMobile"}
+    )
+    public void adsVuapVideoClosesWhenTapCloseButtonFandomMobile(String pageType, String pageName, String slotName, String iframeId) {
+        AdsFandomObject fandomPage = loadPage(pageName, pageType);
+        VideoFanTakeover videoFanTakeover = prepareSlot(slotName, iframeId, fandomPage);
+
+        videoFandomPage().verifyVideoClosesAfterTapOnCloseButton(slotName, videoFanTakeover);
+    }
+
+    @NetworkTrafficDump
+    @InBrowser(
+            browser = Browser.CHROME,
+            emulator = Emulator.GOOGLE_NEXUS_5
+    )
+    @Test(
+            dataProviderClass = FandomAdsDataProvider.class,
+            dataProvider = "fandomVuapPageMobile",
+            groups = {"AdsVuapFandomMobile", "AdsVuapTimeProgressingFandomMobile"}
+    )
+    public void adsVuapTimeProgressingFandomMobile(String pageType, String pageName, String slotName, String iframeId) throws InterruptedException {
+        networkTrafficInterceptor.startIntercepting();
+        AdsFandomObject fandomPage = loadPage(pageName, pageType);
+        VideoFanTakeover videoFanTakeover = prepareSlot(slotName, iframeId, fandomPage);
+
+        videoFandomPage().verifyIsVideoTimeProgresingOnMobile(slotName, networkTrafficInterceptor, videoFanTakeover);
+    }
+
+    @NetworkTrafficDump
+    @InBrowser(
+            browser = Browser.CHROME,
+            emulator = Emulator.GOOGLE_NEXUS_5
+    )
+    @Test(
+            dataProviderClass = FandomAdsDataProvider.class,
+            dataProvider = "fandomVuapPageMobile",
+            groups = {"AdsVuapFandomMobile", "AdsVuapVideoPauseFandomMobile"}
+    )
+    public void adsVuapVideoPauseFandomMobile(String pageType, String pageName, String slotName, String iframeId) throws InterruptedException {
+        networkTrafficInterceptor.startIntercepting();
+        AdsFandomObject fandomPage = loadPage(pageName, pageType);
+        VideoFanTakeover videoFanTakeover = prepareSlot(slotName, iframeId, fandomPage);
+
+        videoFandomPage().verifyIsVideoPausedOnMobile(slotName, networkTrafficInterceptor, videoFanTakeover);
+    }
+
+    private VideoFanTakeover prepareSlot(String slotName, String iframeId, AdsFandomObject fandomPage) {
+        fandomPage.triggerOnScrollSlots();
+        VideoFanTakeover videoFanTakeover = new VideoFanTakeover(driver, iframeId);
+        fandomPage.scrollToSlot(AdsFandomContent.getSlotSelector(slotName));
+        return videoFanTakeover;
+    }
+
+    private FandomVideoFanTakeover videoFandomPage() {
+        FandomVideoFanTakeover videoFandomPage = new FandomVideoFanTakeover(driver);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "videoFandomPage". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1488)